src/Type/ArrayType.php

```diff
@@ -38,6 +38,11 @@ public function __construct(
 		$this->callable = $callable ?? TrinaryLogic::createMaybe()->and((new StringType)->isSuperTypeOf($itemType));
 	}
 
+	public function getKeyType(): Type
+	{
+		return $this->keyType;
```

Comment:
Is getKeyType and getItemType necessary if we have `getIterableKeyType`/`getIterableValueType `? 🤔 If you still think that yes, there are places in the codebase that would be suited better to be using `getKeyType`, should be modified in this commit.